src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -230,7 +230,7 @@ def __repr__(self) -> str:
         txt = f"{clsname}: {nchan} channels - {nunits} units - {nseg} segments - {self.format}"
         if self.is_sparse():
             txt += " - sparse"
-        if self.has_recording():
+        if self.has_recording() or self.has_temporary_recording():
```

Comment:
One tiny idea (but more for discussion not a hard suggestion). Would it be beneficial to update the repr such that it says whether it is `has recording` or `has temp recording` so the user has a quick display to remind them if this was the original that made the object or a temp they just attached? Might be overkill, but I could imagine wanting to double check if I were trying to reanalyze a bunch of datasets some with original and some where I attached a temp.